quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/WithDatabase.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.it.junit5.extension;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+
+/**
+ * {@link WithDatabase} is used to start a database container.
+ */
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@EnabledIfSystemProperty(named = "kc.test.storage.database", matches = "true", disabledReason = "Docker takes too much time and stability depends on the environment. We should try running these tests in CI but isolated.")
```

Comment:
I don't think this explanation is fair:

> Docker takes too much time and stability depends on the environment.

- re: `too much time` is now about 10 minutes in CI vs. 7/8 minutes of the Raw distribution
- re: `stability depends on the environment` in CI and in development machines it's very much stable

Reply:
I did not try yet to run all database tests. Locally they are not very stable for me and I would gradually add more database tests.

Reply:
FYI, we really lack GHA resources. Unfortunately, we can not have everything we want. I'm even seeing errors related to surefire not being able to fork processes, completely unrelated to any change.

Also, these tests require a higher JVM heap/metaspace.

What we really need to solve is how to get more gears ...

Reply:
Here I still need to ask you to rephrase or even remove the `disabledReason`